src/execution_tree/compiler/compiler.cpp

```diff
@@ -343,12 +331,13 @@ namespace phylanx { namespace execution_tree { namespace compiler
             std::string name = ast::detail::identifier_name(name_expr);
 
             // get global name of the component created
-            std::string full_name = name;
+            primitive_name_parts name_parts;
+            name_parts.instance = name;
+
             ast::tagged id = ast::detail::tagged_id(name_expr);
-            if (id.id >= 0)
-            {
-                full_name += annotation(id);
-            }
+            name_parts.compile_id = snippets_.compile_id_ - 1;
+            name_parts.tag1 = id.id;
+            name_parts.tag2 = id.col;
```

Comment:
Please use the constructor of `primitive_name_parts`.

Reply:
Can't. The value of `primitive`, which is needed in the ctor, is not decided yet